From b98ed87225fa3c0f156b0798ca5c285d8fa8fac1 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Tue, 15 Jun 2021 07:27:41 +0200
Subject: [PATCH] XMLParser.cpp: include string.h

Include string.h to avoid the following build failure:

/home/buildroot/autobuild/run/instance-3/output-1/build/cegui-00b4e1fe174da53b7ed726ab5970ba51bd5b5ee0/cegui/src/XMLParserModules/RapidXML/XMLParser.cpp:73:5: error: 'memcpy' was not declared in this scope
   73 |     memcpy(buf, source.getDataPtr(), size);
      |     ^~~~~~

Fixes:
 - http://autobuild.buildroot.org/results/1cb09e5f52435efc505c61707b2d5d2ee871524b

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Upstream: 628f7edf4b420f752f1965ea11e5e747e3ff93b0
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
---
 cegui/src/XMLParserModules/RapidXML/XMLParser.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cegui/src/XMLParserModules/RapidXML/XMLParser.cpp b/cegui/src/XMLParserModules/RapidXML/XMLParser.cpp
index 810ea6bbf..59822fc80 100644
--- a/cegui/src/XMLParserModules/RapidXML/XMLParser.cpp
+++ b/cegui/src/XMLParserModules/RapidXML/XMLParser.cpp
@@ -38,6 +38,8 @@
 #include "CEGUI/Logger.h"
 #include "CEGUI/Exceptions.h"
 
+#include <string.h>
+
 // Start of CEGUI namespace section
 namespace CEGUI
 {
-- 
2.31.1

